test/api/profile.test.js

```diff
@@ -0,0 +1,98 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+import supertest from 'supertest';
+import { Storage as MockStorage } from '../../__mocks__/@google-cloud/storage';
+import { createApp } from '../../src/app';
+import { ACCEPT_VALUE_MIME } from '../../src/middlewares/versioning';
+import { generateToken, decodeToken } from '../../src/logic/jwt';
+
+beforeEach(() => MockStorage.cleanUp());
+afterEach(() => MockStorage.cleanUp());
+
+describe('DELETE /profile', () => {
+  function setup() {
+    const acceptHeader = ACCEPT_VALUE_MIME + ';version=1';
+    const agent = supertest(createApp().callback());
+    return { acceptHeader, agent };
+  }
+
+  it('gives a 200 response when successfully uploading a profile', async function() {
+    const { agent, acceptHeader } = setup();
+
+    // First send a request to create a file.
+    const result = await agent
+      .post('/compressed-store')
+      .accept(acceptHeader)
+      .type('text')
+      .send('SOME_RANDOM_CONTENT')
+      .expect(200);
+
+    if (!result.text) {
+      // Flow couldn't refine this on its own:
+      throw new Error(
+        'Could not find the text response from the post to compressed store.'
+      );
+    }
+    const jwtToken = result.text;
+    const { profileToken } = (decodeToken(jwtToken): any);
+
+    // Then delete the file.
+    await agent
+      .delete(`/profile/${profileToken}`)
+      .accept(acceptHeader)
+      .set('Authorization', `Bearer ${jwtToken}`)
+      .send()
+      .expect(200)
+      .expect('Profile successfully deleted.');
+
+    // Sending it another time will result in a 400 for it not existing.
+    await agent
+      .delete(`/profile/${profileToken}`)
+      .accept(acceptHeader)
+      .set('Authorization', `Bearer ${jwtToken}`)
+      .send()
+      .expect(400)
+      .expect('That profile was most likely already deleted.');
```

Comment:
optional: because there's another test for the "not found" case, I believe that here you're mostly testing the mock (testing it returns 404).
Maybe checking `MockStorage.buckets[XXX].files[XXX]` exists before, and doesn't exist after, would be better.